Add Playwright testing framework and initial test cases for the CMS

MrDirkelz opened the pull request.

MrDirkelz committed:
Add Playwright testing framework and initial test cases for the CMS

MrDirkelz committed:
Rename Playwright test workflows for clarity

MrDirkelz committed:
Update package.json to include additional dependencies and improve script organization

MrDirkelz committed:
Add Auth0 login helper function for Playwright tests

MrDirkelz committed:
Comment out Auth0 login helper function in Playwright tests

MrDirkelz committed:
Refactor Auth0 login helper function and comment out unused browser configurations in Playwright setup

MrDirkelz committed:
Refactor Auth0 login handling by removing the old login function and implementing a new user creation flow in Playwright tests

MrDirkelz committed:
Add tests for user login and post creation in the CMS client

DirkDevelopmentFreelance commented:
Status:

1. Local tests have been set up and are working
2. Specific Auth0 instance has been set up for end to end tests

Todo:
- [ ] Set up github env variables set specifically for end to end tests.
- [ ] Ensure timings are correct for github actions and how slow it is.
- [ ] Ensure API is running correctly, alongside user access has to be set to super-admin for the e2e test user

MrDirkelz committed:
Update test cases and fix document creation logic in Playwright tests

MrDirkelz committed:
Update Playwright configuration and tests to use port 5174

MrDirkelz committed:
Refactor environment variable setup in Playwright workflow for API and app

MrDirkelz committed:
Add temporary cms directory to workflow trigger paths in Playwright tests to test App works with new env vars

MrDirkelz committed:
Add environment context to API and app setup steps in Playwright workflow

MrDirkelz committed:
Fix working-directory assignments for API and app environment variable setup in Playwright workflow

MrDirkelz committed:
Refactor Playwright workflow to separate API and app jobs, streamline environment variable setup, and improve readability

MrDirkelz committed:
Add missing environment variable setup for JWT_SECRET and S3_ENDPOINT in API workflow

MrDirkelz committed:
Reorder Setup API step in workflow to ensure environment variables are set before installation

MrDirkelz committed:
Remove debug echo for DB_CONNECTION_STRING and add S3_ENDPOINT setup in API environment variables

MrDirkelz committed:
Fix working-directory assignments for API and app environment variable setup in Playwright workflow

MrDirkelz committed:
Refactor environment variable setup in API and app workflows to enhance clarity and include additional variables

MrDirkelz committed:
Reorganize setup steps for app and shared library in Playwright workflow

MrDirkelz committed:
Comment out favicon link in index.html for future reference

MrDirkelz committed:
Uncomment favicon link in index.html for future use

MrDirkelz committed:
Reduce test timeout from 100 seconds to 12 seconds for improved efficiency

MrDirkelz committed:
Add VITE_LOGO_FAVICON environment variable setup in app workflow

MrDirkelz committed:
Add VITE_LOGO_FAVICON to environment variables setup in app workflow

MrDirkelz committed:
Fix environment variable reference for VITE_LOGO_FAVICON in app-playwright.yml

MrDirkelz committed:
Remove debug print statement for VITE_LOGO_FAVICON in app-playwright.yml

MrDirkelz committed:
Add installation of browser dependencies in app workflow

MrDirkelz committed:
Refactor Playwright setup by removing manual browser dependency installation in favor of using Playwright GitHub Action

MrDirkelz committed:
Refactor Playwright setup to use --with-deps flag for browser installation

MrDirkelz committed:
Refactor Playwright workflow: consolidate jobs, update environment variable references, and enhance timeout settings

MrDirkelz committed:
Increase timeout for e2e tests from 10 to 20 minutes

MrDirkelz committed:
Increase sleep duration for API startup from 20 to 50 seconds

DirkDevelopmentFreelance commented:
Hi @ivanslabbert, like last time I got the end to end tests to work locally, and I am trying to get the app to also use the new auth0 end-to-end environment that was created. But I am struggling to get everything working in github actions. I have also tried running it in codespaces but I can't figure out how to get it to work in github actions. I was wondering if you could maybe help me on that part? As it is mostly not related to the tests

I also tried looking at how someone else did it, but I just can't seem to crack the github actions side

Here is a reference to somebody else's PR for the end to end testing, but I don't see that they are using it like we want to use it
https://github.com/onetimesecret/onetimesecret/pull/710/files#diff-c5d401e588c34e31052f1b0c43179b5bc943d595859afbe9f21ecc35fa84620a

MrDirkelz committed:
Fix: Update e2e tests for CMS and Auth0 login

- Updated the test for creating a new document.
- Fixed the Auth0 login button selector.
- Removed the failing login test.
- Updated the baseURL in playwright config.
- Updated the create button selector

Fix: Update e2e tests for CMS and Auth0 login

- Updated Auth0 login to click the "accept" button.
- Fixed CMS create new document flow.
- Updated the vite preview port.
- Removed the login test, as login is implicit in other tests.
- Added data-test attributes to the create button.
- Added retry to test

MrDirkelz committed:
Update Playwright to version 1.55.0 in package.json and package-lock.json

MrDirkelz committed:
Fix: Update selector for options button in document creation test

MrDirkelz committed:
Add: Create end-to-end testing workflow for App and CMS using Playwright

MrDirkelz committed:
Remove: Delete CMS Playwright testing workflow

MrDirkelz committed:
Fix: Enhance Node setup and environment variable configuration for end-to-end tests

MrDirkelz committed:
Fix: Update end-to-end tests workflow for CMS and adjust caching settings

MrDirkelz committed:
Fix: Move JWT_SECRET environment variable to the API start step

johan-bell commented:
@DirkDevelopmentFreelance We can pair-program on it if it is needed.
